tests/Foundation/Http/Middleware/CheckForTrustedHostTest.php

```diff
@@ -0,0 +1,145 @@
+{
+    protected static $orignalTrustHosts;
+
+    public static function setUpBeforeClass(): void
+    {
+        self::$orignalTrustHosts = Request::getTrustedHosts();
+    }
+
+    public static function tearDownAfterClass(): void
+    {
+        Request::setTrustedHosts(self::$orignalTrustHosts);
+    }
+
+    public function testTrustedHost()
+    {
+        $trustedHosts = ['octobercms.com'];
+        $headers = ['HOST' => 'octobercms.com'];
+        $urlGenerator = $this->createUrlGenerator($trustedHosts, $headers);
+        $url = $urlGenerator->to('/');
+
+        $this->assertEquals('http://octobercms.com', $url);
+    }
+
+    public function testTrustedHostWwwSubdomain()
+    {
+        $trustedHosts = ['octobercms.com'];
+        $headers = ['HOST' => 'www.octobercms.com'];
+        $urlGenerator = $this->createUrlGenerator($trustedHosts, $headers);
+        $url = $urlGenerator->to('/');
+
+        $this->assertEquals('http://www.octobercms.com', $url);
+    }
+
+    public function testTrustedIpHost()
+    {
+        $trustedHosts = ['127.0.0.1'];
+        $headers = ['HOST' => '127.0.0.1'];
+        $urlGenerator = $this->createUrlGenerator($trustedHosts, $headers);
+        $url = $urlGenerator->to('/');
+
+        $this->assertEquals('http://127.0.0.1', $url);
+    }
+
+    public function testNoTrustedHostsSet()
+    {
+        $trustedHosts = false;
+        $headers = ['HOST' => 'malicious.com'];
+        $urlGenerator = $this->createUrlGenerator($trustedHosts, $headers);
+        $url = $urlGenerator->to('/');
+
+        $this->assertEquals('http://malicious.com', $url);
+    }
+
+    public function testTrustedIpHostWwwSubdomain()
+    {
+        $this->expectException(SuspiciousOperationException::class);
+
+        $trustedHosts = ['127.0.0.1'];
+        $headers = ['HOST' => 'www.127.0.0.1'];
+        $urlGenerator = $this->createUrlGenerator($trustedHosts, $headers);
+        $url = $urlGenerator->to('/');
+    }
+
+    public function testThrowExceptionForUntrustedHosts()
+    {
+        $this->expectException(SuspiciousOperationException::class);
+
+        $trustedHosts = ['octobercms.com'];
+        $headers = ['HOST' => 'malicious.com'];
+        $urlGenerator = $this->createUrlGenerator($trustedHosts, $headers);
+        $urlGenerator->to('/');
+    }
+
+    public function testThrowExceptionForUntrustedServerName()
+    {
+        $this->expectException(SuspiciousOperationException::class);
+
+        $trustedHosts = ['octobercms.com'];
+        $headers = [];
+        $servers = ['SERVER_NAME' => 'malicious.com'];
+        $urlGenerator = $this->createUrlGenerator($trustedHosts, $headers, $servers);
+        $urlGenerator->to('/');
+    }
+
+    public function testThrowExceptionForUntrustedServerAddr()
+    {
+        $this->expectException(SuspiciousOperationException::class);
+
+        $trustedHosts = ['octobercms.com'];
+        $headers = [];
+        $servers = ['SERVER_NAME' => 'malicious.com'];
```

Comment:
This looks identical to the untrusted server name test

Reply:
Whoops, that's supposed to be SERVER_ADDR - fixed.